core/drivers/unix_socket.cc

```diff
@@ -38,53 +41,63 @@
 // TODO: Revise this once the interrupt mode is  implemented.
 
 #define RECV_SKIP_TICKS 256
-#define MAX_TX_FRAGS 8
+#define SIG_THREAD_EXIT SIGUSR2
 
-void UnixSocketPort::AcceptNewClient() {
-  int ret;
 
-  for (;;) {
-    ret = accept4(listen_fd_, nullptr, nullptr, SOCK_NONBLOCK);
-    if (ret >= 0) {
-      break;
-    }
+void UnixSocketPort::AcceptThread() {
+  sigset_t sigset;
+  sigfillset(&sigset);
+  sigdelset(&sigset, SIG_THREAD_EXIT);
 
-    if (errno != EINTR) {
-      PLOG(ERROR) << "[UnixSocket]:accept4()";
-    }
-  }
+  struct pollfd fds[2];
+  memset(fds, 0, sizeof(fds));
+  fds[0].fd = listen_fd_;
+  fds[0].events = POLLIN;
+  fds[1].events = POLLRDHUP;
 
-  recv_skip_cnt_ = 0;
+  while (true) {
+    fds[1].fd = client_fd_;
+    int res = ppoll(fds, 2, nullptr, &sigset);
 
-  if (old_client_fd_ != kNotConnectedFd) {
-    // Reuse the old file descriptor number by atomically exchanging the new fd
-    // with the
-    // old one.  The zombie socket is closed silently (see dup2).
-    dup2(ret, client_fd_);
-    close(ret);
-  } else {
-    client_fd_ = ret;
-  }
-}
+    if (accept_thread_stop_req_) {
+      return;
 
-// This accept thread terminates once a new client is connected.
-void *AcceptThreadMain(void *arg) {
-  UnixSocketPort *p = reinterpret_cast<UnixSocketPort *>(arg);
-  p->AcceptNewClient();
-  return nullptr;
+    } else if (res < 0) {
+      if (errno == EINTR) {
+        continue;
+      } else {
+        PLOG(ERROR) << "[UnixSocketPort]:epoll_wait()";
```

Comment:
Feel free to ignore this: you don't need to add `[UnixSocketPort]:` to the log message since the filename will be automatically added by `PLOG`. Also, `epoll_wait()` -> `ppoll()`